bundles/org.openhab.core.ephemeris/src/main/java/org/eclipse/smarthome/core/ephemeris/internal/EphemerisManagerImpl.java

```diff
@@ -0,0 +1,228 @@
+ * others.
+ *
+ * @author Gaël L'hopital - Initial contribution and API
+ */
+@Component(immediate = true, name = "org.openhab.ephemeris", property = {
+        Constants.SERVICE_PID + "=org.openhab.ephemeris", ConfigurableService.SERVICE_PROPERTY_CATEGORY + "=system",
+        ConfigurableService.SERVICE_PROPERTY_LABEL + "=Ephemeris",
+        ConfigurableService.SERVICE_PROPERTY_DESCRIPTION_URI + "=" + EphemerisManagerImpl.CONFIG_URI })
+@NonNullByDefault
+public class EphemerisManagerImpl implements EphemerisManager, ConfigOptionProvider {
+    private final Logger logger = LoggerFactory.getLogger(EphemerisManagerImpl.class);
+
+    // constants for the configuration properties
+    protected static final String CONFIG_URI = "system:ephemeris";
+    private static final String CONFIG_DAYSET_PREFIX = "dayset-";
+    private static final String CONFIG_DAYSET_WEEKEND = "weekend";
+    private static final String CONFIG_COUNTRY = "country";
+    private static final String CONFIG_REGION = "region";
+    private static final String CONFIG_CITY = "city";
+
+    private final Map<String, Set<DayOfWeek>> daysets = new HashMap<>();
+    private final Map<Object, HolidayManager> holidayManagers = new HashMap<>();
+    private final List<String> countryParameters = new ArrayList<>();
+
+    private @NonNullByDefault({}) String country;
+
+    private @NonNullByDefault({}) LocaleProvider localeProvider;
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        modified(config);
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        config.entrySet().stream().filter(e -> e.getKey().startsWith(CONFIG_DAYSET_PREFIX)).forEach(e -> {
+            String[] setDefinition = e.getValue().toString().toUpperCase().split(",");
+            String[] setNameParts = e.getKey().split("-");
+            if (setDefinition.length > 0 && setNameParts.length > 1) {
+                Set<DayOfWeek> dayset = new HashSet<>();
+                Stream.of(setDefinition).forEach(day -> {
+                    dayset.add(DayOfWeek.valueOf(day));
+                });
+                daysets.put(setNameParts[1], dayset);
+            } else {
+                logger.warn("Erroneous dayset definition {} : {}", e.getKey(), e.getValue());
+            }
+        });
+
+        if (config.containsKey(CONFIG_COUNTRY)) {
+            country = config.get(CONFIG_COUNTRY).toString();
+        } else {
+            country = localeProvider.getLocale().getCountry();
+            logger.debug("Using system default country '{}' ", country);
+        }
+
+        if (config.containsKey(CONFIG_REGION)) {
+            countryParameters.add(config.get(CONFIG_REGION).toString());
+        }
+
+        if (config.containsKey(CONFIG_CITY)) {
+            countryParameters.add(config.get(CONFIG_CITY).toString());
+        }
+    }
+
+    @Reference
+    protected void setLocaleProvider(LocaleProvider localeProvider) {
```

Comment:
Can be simplified and the nulnness workaround an be removed if constructor injection is used (using constructor injection is not a must but I assume we all like to make the code better).